### PR TITLE
splunkclient-go: Do not use incorrect build tags

### PR DESCRIPTION
**Pull request by andrewwdye**
## Overview
`go build` (and friends) include build tags for current and prior versions of go (i.e., go 1.21 has tags go1.21, go1.20, go1.19, ...). The existing `go:build` directive in splunkclient-go checks the for absence of tags go1.1 -> go1.16, which will always fail.

This change modifies the directive to indicate go 1.17 as the min version.

Also fixed lint and test failures. For the latter, removed attributed that were tested for but [not returned](https://github.com/open-telemetry/opentelemetry-go/blob/87396747adaad1d510214d14d5afc386652f8bdf/semconv/v1.17.0/httpconv/http.go#L76-L81) from `ClientRequest` for semconv 1.17.

## Testing
Before
```
❯ cd splunk-otel-go/instrumentation/k8s.io/client-go/splunkclient-go
❯ go test ./... -v
?   	github.com/signalfx/splunk-otel-go/instrumentation/k8s.io/client-go/splunkclient-go	[no test files]
?   	github.com/signalfx/splunk-otel-go/instrumentation/k8s.io/client-go/splunkclient-go/internal/config	[no test files]
?   	github.com/signalfx/splunk-otel-go/instrumentation/k8s.io/client-go/splunkclient-go/option	[no test files]
?   	github.com/signalfx/splunk-otel-go/instrumentation/k8s.io/client-go/splunkclient-go/transport	[no test files]
```

After
```
❯ go test ./... -v
?   	github.com/signalfx/splunk-otel-go/instrumentation/k8s.io/client-go/splunkclient-go	[no test files]
?   	github.com/signalfx/splunk-otel-go/instrumentation/k8s.io/client-go/splunkclient-go/internal/config	[no test files]
?   	github.com/signalfx/splunk-otel-go/instrumentation/k8s.io/client-go/splunkclient-go/option	[no test files]
=== RUN   TestNewWrapperFuncNilRoundTripper
--- PASS: TestNewWrapperFuncNilRoundTripper (0.00s)
=== RUN   TestRequestToSpanNameUnrecognized
--- PASS: TestRequestToSpanNameUnrecognized (0.00s)
=== RUN   TestRequestPathToSpanName
--- PASS: TestRequestPathToSpanName (0.00s)
=== RUN   TestRequestMethodToSpanName
--- PASS: TestRequestMethodToSpanName (0.00s)
=== RUN   TestWrappedBodyRead
--- PASS: TestWrappedBodyRead (0.00s)
=== RUN   TestWrappedBodyReadEOFError
--- PASS: TestWrappedBodyReadEOFError (0.00s)
=== RUN   TestWrappedBodyReadError
--- PASS: TestWrappedBodyReadError (0.00s)
=== RUN   TestWrappedBodyClose
--- PASS: TestWrappedBodyClose (0.00s)
=== RUN   TestWrappedBodyCloseError
--- PASS: TestWrappedBodyCloseError (0.00s)
=== RUN   TestWrappedRoundTripperError
--- PASS: TestWrappedRoundTripperError (0.00s)
PASS
ok  	github.com/signalfx/splunk-otel-go/instrumentation/k8s.io/client-go/splunkclient-go/transport	(cached)
```

fixes https://github.com/signalfx/splunk-otel-go/issues/2890

### CODE REVIEW

**Comment by andrewwdye** [comment]
I have read the CLA Document and I hereby sign the CLA

**Comment by andrewwdye** [comment]
recheck

File: instrumentation/k8s.io/client-go/splunkclient-go/option/option.go [hunk 1]
```diff
@@ -12,8 +12,8 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-//go:build !(go1.1 || go1.2 || go1.3 || go1.4 || go1.5 || go1.6 || go1.7 || go1.8 || go1.9 || go1.10 || go1.11 || go1.12 || go1.13 || go1.14 || go1.15 || go1.16)
-// +build !go1.1,!go1.2,!go1.3,!go1.4,!go1.5,!go1.6,!go1.7,!go1.8,!go1.9,!go1.10,!go1.11,!go1.12,!go1.13,!go1.14,!go1.15,!go1.16
+//go:build go1.17
+// +build go1.17
```
Comment (pellared) [inline instrumentation/k8s.io/client-go/splunkclient-go/option/option.go:16]: Maybe we can simply remove this lines? 

Take notice: https://github.com/signalfx/splunk-otel-go/blob/cdf7850435de19172273e62228b6ef1f49dc2580/instrumentation/k8s.io/client-go/splunkclient-go/go.mod#L3

It looks like a legacy code.

**Review by pellared — COMMENTED** [review]
@andrewwdye Thanks for the issue and PR 👍 

Can you also try adding a changelog entry to make the PR more complete?

File: instrumentation/k8s.io/client-go/splunkclient-go/option/option.go (hunk 1 above)
Comment (andrewwdye) [reply instrumentation/k8s.io/client-go/splunkclient-go/option/option.go]: Sounds reasonable, done

File: CHANGELOG.md [hunk 2]
```diff
@@ -8,6 +8,11 @@ This project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.htm
 
 ## [Unreleased]
 
+### Fixed
+
+- Drop errant build tags in `splunkclient-go`, allowing import with go 1.20+. Also fixes 
+  lint and test failures that were previously skipped.
```
Comment (pellared) [inline CHANGELOG.md:14]:
```suggestion
- Drop errant build tags in `splunkclient-go`, allowing import with Go 1.20+. Also fixes 
  lint and test failures that were previously skipped. (#2891)
```

**Review by pellared — APPROVED** [review]
@MrAlias, PTAL

**Review by MrAlias — APPROVED** [review]

**Comment by MrAlias** [comment]
@andrewwdye thank you for the contribution! Please [sign your commits](https://docs.github.com/en/authentication/managing-commit-signature-verification/signing-commits). We required signed commits for the project to ensure authenticity.

**Comment by andrewwdye** [comment]
I rebased with my commits signed (and @pellared 's commits signed as co-authored). I assume there is another step here to co-sign but was unsure

**Pull request merged**